fnet/network.go

```diff
@@ -34,6 +35,91 @@ func NormalizePort(port string) string {
 	return ":" + port
 }
 
+// BracketizeIPv6Address returns s in brackets if s is an IPv6 address.
+func BracketizeIPv6Address(s string) string {
+	ip := net.ParseIP(s)
+	switch {
+	case ip == nil:
+		// s is not a valid ip.
+		log.Errf("Invalid IP address: %s", s)
+		return s
+	case ip.To4() != nil:
+		// s is an IPv4 address, do not wrap s in brackets.
+		log.Errf("Address is IPv4, not wrapping %s in brackets.", s)
+		return s
+	default:
+		// s must be an IPv6 address, so wrap s in brackets.
+		log.Infof("Address is IPv6, wrapping %s in brackets.", s)
+		return "[" + s + "]"
+	}
+}
+
+// AppendPort parses s as a url and returns s with the host-portion of s as s:port.
+// s is returned unmodified if s cannot be parsed or the host-portion of s is an
+// invalid domain name or invalid IP address.
+func AppendPort(s string) string {
+	u, err := url.Parse(s)
+	if err != nil {
+		log.Errf("Unable to Parse URL %s: %v", err)
+		return s
+	}
+	var port string
+	switch {
+	case u.Scheme == "http":
+		port = "80"
+	case u.Scheme == "https":
+		port = "443"
+	default:
+		// Unsupported scheme for valid url s
+		log.Errf("Unsupported URL scheme: %s", u.Scheme)
+		return s
+	}
+	ip := net.ParseIP(u.Host)
```

Comment:
```
$ git grep -n net.ParseIP
fgrpc/grpcrunner.go:188:        if ip := net.ParseIP(dest); ip != nil {
fnet/network.go:77:     isAddr := net.ParseIP(host)
```
let's clean up the duplicate if possible
(and if the  BracketizeIPv6Address was needed we should parse only once)

Reply:
OK. When I wrote this, I didn't realize it would be just for grpc url parsing. Let me try to have the url parsing func utilize the grpcDestination func.